# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by scripts/Create-CopyPatch.
# 
# T2 SDE: package/*/gcc/sparcv789.patch
# Copyright (C) 2020 - 2021 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- gcc-10.2.0/gcc/config.gcc.vanilla	2020-10-03 12:59:15.994020808 +0200
+++ gcc-10.2.0/gcc/config.gcc	2020-10-03 13:00:44.134020333 +0200
@@ -3288,7 +3288,7 @@
 	tm_file="${tm_file} dbxelf.h elfos.h sparc/sysv4.h sparc/sp-elf.h sparc/rtemself.h rtems.h newlib-stdint.h"
 	tmake_file="${tmake_file} sparc/t-sparc sparc/t-rtems"
 	;;
-sparc-*-linux*)
+sparc-*-linux*|sparcv*-*-linux*)
 	tm_file="${tm_file} dbxelf.h elfos.h sparc/sysv4.h gnu-user.h linux.h glibc-stdint.h sparc/tso.h"
 	extra_options="${extra_options} sparc/long-double-switch.opt"
 	case ${target} in
@@ -5057,7 +5057,7 @@
 		for which in cpu cpu_32 cpu_64 tune tune_32 tune_64; do
 			eval "val=\$with_$which"
 			case ${val} in
-			"" | sparc | sparcv9 | sparc64 \
+			"" | sparc | sparcv7 | sparcv8 | sparcv9 | sparc64 \
 			| v7 | cypress \
 			| v8 | supersparc | hypersparc | leon | leon3 | leon3v7 \
 			| sparclite | f930 | f934 | sparclite86x \
--- gcc-10.2.0/libgcc/config.host.vanilla	2020-10-03 14:28:28.829991995 +0200
+++ gcc-10.2.0/libgcc/config.host	2020-10-03 14:28:52.561991867 +0200
@@ -1363,7 +1363,7 @@
 	tmake_file="${tmake_file} t-fdpbit t-crtfm"
 	extra_parts="$extra_parts crti.o crtn.o crtfastmath.o"
 	;;
-sparc-*-linux*)		# SPARC's running GNU/Linux, libc6
+sparc-*-linux*|sparcv*-*-linux*)		# SPARC's running GNU/Linux, libc6
 	tmake_file="${tmake_file} t-crtfm"
 	if test "${host_address}" = 64; then
 		tmake_file="$tmake_file sparc/t-linux64"
